src/platform/Zephyr/WiFiManager.h

```diff
@@ -41,9 +41,50 @@ using WpaNetwork = struct wpa_ssid;
 namespace chip {
 namespace DeviceLayer {
 
-class WiFiManager
+// emulation of dictionary - might be moved to utils
+template <typename T1, typename T2, std::size_t N>
+class Map
 {
+    struct Pair
+    {
+        T1 key;
+        T2 value;
+    };
+
+public:
+    Map(const Pair (&list)[N])
+    {
+        int idx{ 0 };
+        for (const auto & pair : list)
+        {
+            mMap[idx++] = pair;
+        }
+    }
+
+    T2 operator[](const T1 & key) const
+    {
+        for (const auto & it : mMap)
+        {
+            if (key == it.key)
+                return it.value;
+        }
+
+        return T2{};
+    }
+
+    Map()            = delete;
+    Map(const Map &) = delete;
+    Map(Map &&)      = delete;
+    Map & operator=(const Map &) = delete;
+    Map & operator=(Map &&) = delete;
+    virtual ~Map()          = default;
```

Comment:
We don't need `virtual`

Reply:
I agree, for now we don't. But we do not prevent this class from being inherited too. So in case it is, isn't it safer to ensure virtual d'tor is here?

Reply:
Idk, as long as we don't have any virtual methods, I don't see a value in adding a virtual destructor and waste flash for vtables. STL containers also don't provide virtual destructors although one can inherit from them.

Reply:
Cruel embedded world...